programs/clearing_house/src/controller/liquidation/tests.rs

```diff
@@ -1860,3 +1860,301 @@ pub mod liquidate_perp_pnl_for_deposit {
+            initialized: true,
+            liquidation_fee: LIQUIDATION_FEE_PRECISION / 100,
+            ..Market::default()
+        };
+        create_anchor_account_info!(market, Market, market_account_info);
+        let market_map = MarketMap::load_one(&market_account_info, true).unwrap();
+
+        let mut bank = Bank {
+            bank_index: 0,
+            oracle_source: OracleSource::QuoteAsset,
+            cumulative_deposit_interest: BANK_CUMULATIVE_INTEREST_PRECISION,
+            decimals: 6,
+            initial_asset_weight: BANK_WEIGHT_PRECISION,
+            ..Bank::default()
+        };
+        create_anchor_account_info!(bank, Bank, bank_account_info);
+        let bank_map = BankMap::load_one(&bank_account_info, true).unwrap();
+
+        let mut user = User {
+            orders: get_orders(Order {
+                market_index: 0,
+                status: OrderStatus::Open,
+                order_type: OrderType::Limit,
+                direction: PositionDirection::Long,
+                base_asset_amount: BASE_PRECISION,
+                ts: 0,
+                slot: 0,
+                ..Order::default()
+            }),
+            positions: get_positions(MarketPosition {
+                market_index: 0,
+                base_asset_amount: 0,
+                quote_asset_amount: -100 * QUOTE_PRECISION_I128,
+                open_orders: 1,
+                open_bids: BASE_PRECISION_I128,
+                ..MarketPosition::default()
+            }),
+            bank_balances: [UserBankBalance::default(); 8],
+            bankrupt: true,
+            being_liquidated: false,
+            next_liquidation_id: 2,
+            ..User::default()
+        };
+
+        let mut liquidator = User {
+            bank_balances: get_bank_balances(UserBankBalance {
+                bank_index: 0,
+                balance_type: BankBalanceType::Deposit,
+                balance: 50 * BANK_INTEREST_PRECISION,
+            }),
+            ..User::default()
+        };
+
+        let user_key = Pubkey::default();
+        let liquidator_key = Pubkey::default();
+
+        let mut expected_user = user;
+        expected_user.being_liquidated = false;
+        expected_user.bankrupt = false;
+        expected_user.positions[0].quote_asset_amount = 0;
+
+        let mut expected_market = market;
+        expected_market.amm.cumulative_funding_rate_long = 1010 * FUNDING_RATE_PRECISION_I128;
+        expected_market.amm.cumulative_funding_rate_short = -1010 * FUNDING_RATE_PRECISION_I128;
+
+        resolve_perp_bankruptcy(
+            0,
+            &mut user,
+            &user_key,
+            &mut liquidator,
+            &liquidator_key,
+            &market_map,
+            &bank_map,
+            &mut oracle_map,
+            now,
+        )
+        .unwrap();
+
+        assert_eq!(expected_user, user);
+        assert_eq!(expected_market, *market_map.get_ref(&0).unwrap());
```

Comment:
note: good to add two more users here (one long,one short) who 
1. settle funding 
2. assert bankruptcy loss adds up to their funding loss.

Reply:
good idea